### ISSUE TITLE
PRs opened by the `github-actions` bot don't trigger status checks

### ISSUE THREAD

**Issue by jonathan-innis** [aws/karpenter#4525]
### Description

Currently, if a PR is opened by the `github-actions` bot, status checks will just sit in an `Expected` state forever. We need to look into why this automation is broken.

The current workaround is to push a new commit on the branch yourself or to close and re-open the PR to re-trigger the checks.

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by aman0408** [comment]
Can I work on this issue?

**Comment by aman0408** [comment]
This behviour is expected as [described in the documentation](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow)

> When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN, with the exception of workflow_dispatch and repository_dispatch, will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs. For example, if a workflow run pushes code using the repository's GITHUB_TOKEN, a new workflow will not run even when the repository contains a workflow configured to run when push events occur.

**Comment by jonathan-innis** [comment]
> This behviour is expected

Good point. I wonder if there's a different way that we could trigger workflow runs so that we don't have to manually intervene every time one of these PRs is opened.

**Comment by aman0408** [comment]
@jonathan-innis can we use this workaround?
https://github.com/peter-evans/create-pull-request/issues/48#issuecomment-536204092

**Comment by jonathan-innis** [comment]
> can we use this workaround

We need to check with our security team before we could take something like this in the project. Generally, long-term static credentials are frowned upon by security teams because they allow persistent access to the project if they are compromised. We _might_ be able to get around the concern here if the scope of the token is _only_ to open pull requests against the repo. I can check back here with the security team and see what they recommend here.

**Comment by jonathan-innis** [comment]
@aman0408 Checking in here. Are you still planning to work on this one?

**Comment by aman0408** [comment]
Yes @jonathan-innis . I have been busy since a couple of weeks but I should be able to make some progress in this week
